discovery/gossiper.go

```diff
@@ -2454,23 +2454,10 @@ func (d *AuthenticatedGossiper) updateChannel(info *channeldb.ChannelEdgeInfo,
 	}
 	edge.LastUpdate = time.Unix(timestamp, 0)
 
-	chanUpdate := &lnwire.ChannelUpdate{
-		ChainHash:       info.ChainHash,
-		ShortChannelID:  lnwire.NewShortChanIDFromInt(edge.ChannelID),
-		Timestamp:       uint32(timestamp),
-		MessageFlags:    edge.MessageFlags,
-		ChannelFlags:    edge.ChannelFlags,
-		TimeLockDelta:   edge.TimeLockDelta,
-		HtlcMinimumMsat: edge.MinHTLC,
-		HtlcMaximumMsat: edge.MaxHTLC,
-		BaseFee:         uint32(edge.FeeBaseMSat),
-		FeeRate:         uint32(edge.FeeProportionalMillionths),
-		ExtraOpaqueData: edge.ExtraOpaqueData,
-	}
-
 	// With the update applied, we'll generate a new signature over a
 	// digest of the channel announcement itself.
-	sig, err := netann.SignAnnouncement(
+	chanUpdate := netann.UnsignedChannelUpdateFromEdge(info, edge)
```

Comment:
We update the channel update timestamp above, can we use the modifier there as well?

Reply:
yeah don't know i ever went down this path lol